databricks/koalas/config.py

```diff
@@ -298,3 +298,53 @@ def _check_option(key: str) -> None:
         raise OptionError(
             "No such option: '{}'. Available options are [{}]".format(
                 key, ", ".join(list(_options_dict.keys()))))
+
+
+class DictWrapper:
+    """ provide attribute-style access to a nested dict"""
+
+    def __init__(self, d, prefix=""):
+        object.__setattr__(self, "d", d)
+        object.__setattr__(self, "prefix", prefix)
+
+    def __setattr__(self, key, val):
+        prefix = object.__getattribute__(self, "prefix")
+        d = object.__getattribute__(self, "d")
+        if prefix:
+            prefix += "."
+        canonical_key = prefix + key
+
+        candidates = [
+            k for k in d.keys() if all(x in k.split(".") for x in canonical_key.split("."))]
+        if len(candidates) == 1 and candidates[0] == canonical_key:
+            return set_option(canonical_key, val)
+        elif len(candidates) == 0:
+            raise OptionError(
+                "No such option: '{}'. Available options are [{}]".format(
+                    key, ", ".join(list(_options_dict.keys()))))
+        else:
+            return DictWrapper(d, canonical_key)
+
+    def __getattr__(self, key):
+        prefix = object.__getattribute__(self, "prefix")
+        d = object.__getattribute__(self, "d")
+        if prefix:
+            prefix += "."
+        canonical_key = prefix + key
+
+        candidates = [
+            k for k in d.keys() if all(x in k.split(".") for x in canonical_key.split("."))]
+        if len(candidates) == 1 and candidates[0] == canonical_key:
+            return get_option(canonical_key)
+        elif len(candidates) == 0:
+            raise OptionError(
+                "No such option: '{}'. Available options are [{}]".format(
+                    key, ", ".join(list(_options_dict.keys()))))
+        else:
+            return DictWrapper(d, canonical_key)
+
+    def __dir__(self):
+        return list(self.d.keys())
```

Comment:
We might want to filter with prefix and drop the prefix? E.g.,:

```py
>>> dir(ks.options.compute)
['default_index_type', 'max_rows', 'ops_on_diff_frames', 'shortcut_limit']
```

instead of:

```py
>>> dir(ks.options.compute)
['compute.default_index_type', 'compute.max_rows', 'compute.ops_on_diff_frames', 'compute.shortcut_limit', 'display.max_rows', 'plotting.max_rows', 'plotting.sample_ratio']
```

WDYT?

Reply:
Btw, pandas' options only show current child categories:

```py
>>> dir(pd.options)
['compute', 'display', 'html', 'io', 'mode', 'plotting']
>>> dir(pd.options.compute)
['use_bottleneck', 'use_numexpr']
```

I don't think we need to follow this now, but we might want if we have more and more options in the future.

Reply:
Yea sure. Let me think about a better one and fix it today.